plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java

```diff
@@ -179,7 +179,7 @@ private static ZonedDateTime toZonedDateTime(long millisUtc, ZoneId zoneId)
     static String stringToStringConverter(Object value)
     {
         Slice slice = (Slice) value;
-        return quote(slice.toStringUtf8());
+        return quote(slice.toStringUtf8().replace("'", "\\'"));
```

Comment:
It'd be nice to handle all possible characters that need escaping.

If possible can you also add support for the ones listed at https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#string_and_bytes_literals along with test-cases?

Reply:
Sure, let me take a look.

Reply:
@hashhar Adding support for other literals doesn't look straightforward. Is it fine if we go ahead with this one and I'll do some more research and raise a separate issue for other characters?

Reply:
Please create an issue with your findings and add a TODO comment here so that we don't forget.

cc: @ebyhr

Reply:
Sure, I am on it.

Reply:
Added a TODO comment.